libcaf_core/caf/chunked_string.hpp

```diff
@@ -17,25 +18,38 @@
 
 namespace caf {
 
-/// Represents a chunked string that as a linked list of string views.
-struct chunked_string {
-  /// A single node in the linked list.
-  struct node {
-    /// The characters of this chunk.
-    std::string_view content;
+/// Represents a chunked string as a linked list of string views.
+class CAF_CORE_EXPORT chunked_string {
```

Comment:
Whats keeping us from doing an alias here?
```
using chunked_string = detail::json::linked_list<std::string_view> chunks_;
```

Reply:
Two points:

- users mostly care about calling `copy_to` or `copy_quoted_to`, moving these to free functions would make the API more unwieldy
- this type is supposed to be immutable, we don't want to expose `push_back` & friends
- objects of this type are supposed to super small: they are basically the size of single pointer now; a linked list has a lot more going on